### PR TITLE
Remove compile-time dependencies from key tools

### PR DESCRIPTION
**Pull request by danielinux**
1. Remove all compile-time configuration from key tools, to allow for binary distribution of generic, pre-compiled artifacts.

1. Add action to build keytools on windows.
* Update key tools VS project to 2022.
* Fix a few warnings in sign.c
* Fix issues with windows already having min/max.
* Fix (workaround) issue with key tools needing WOLFBOOT_SECTOR_SIZE.

Requires https://github.com/wolfSSL/wolfssl/pull/8257

### CODE REVIEW

**Review by dgarske — DISMISSED** [review]

**Comment by dgarske** [comment]
@danielinux please review workaround for delta.c in sign tools needing to know the sector size (WOLFBOOT_SECTOR_SIZE). Perhaps consider making this an argument or taking in the .config at runtime / arg.

File: include/wolfboot/wolfboot.h [hunk 1]
```diff
@@ -136,17 +138,17 @@ extern "C" {
 /* ML-DSA pub key size is a function of parameters.
  * This needs to be configurable. Default to security
  * category 2. */
```
Comment (philljj) [inline include/wolfboot/wolfboot.h:140]: If we're defaulting to ML_DSA_LEVEL=5 then update or delete this comment please!

**Review by dgarske — APPROVED** [review]

File: tools/keytools/keygen.c [hunk 2]
```diff
@@ -457,15 +451,35 @@ static uint32_t get_pubkey_size(uint32_t keyType)
...
             break;
-#ifdef KEYSTORE_PUBKEY_SIZE_ML_DSA
         case KEYGEN_ML_DSA:
-            size = KEYSTORE_PUBKEY_SIZE_ML_DSA;
+        {
+            char *env_ml_dsa_level = getenv("ML_DSA_LEVEL");
+            if (env_ml_dsa_level == NULL) {
+                fprintf(stderr, "warning: ML_DSA_LEVEL environment variable"
+                       " not set, assuming level 2\n");
+                size = ML_DSA_L2_PUBKEY_SIZE;
```
Comment (bigbrett) [inline tools/keytools/keygen.c:460]: don't we have level 5 set as the default in other areas of the code? E.g. default in the Keytools Makefile and wolfboot.h. Could this ever lead to a weird "silent" mismatch?

File: tools/keytools/sign.c [hunk 3]
```diff
@@ -42,13 +42,10 @@
...
 #include <stddef.h>
 #include <inttypes.h>
-/* target.h is a generated file based on .config (see target.h.in)
- * Provides: WOLFBOOT_SECTOR_SIZE */
-#include <target.h>
 #include <delta.h>
 
 #include "wolfboot/version.h"
-#include "wolfboot/wolfboot.h"
+//#include "wolfboot/wolfboot.h"
```
Comment (bigbrett) [inline tools/keytools/sign.c:48]: dead code

File: tools/keytools/sign.c [hunk 4]
```diff
@@ -344,10 +318,10 @@ static uint16_t sign_tool_find_header(uint8_t *haystack, uint16_t type, uint8_t
 
         len = p[2] | (p[3] << 8);
         /* check len */
-        if ((4 + len) > (uint16_t)(IMAGE_HEADER_SIZE - IMAGE_HEADER_OFFSET)) {
+        if ((4 + len) > (uint16_t)(CMD.header_sz - IMAGE_HEADER_OFFSET)) {
             fprintf(stderr, "This field is too large (bigger than the space available "
                      "in the current header)\n");
-            //fprintf(stderr, "%d %d %d\n", len, IMAGE_HEADER_SIZE, IMAGE_HEADER_OFFSET);
+            //fprintf(stderr, "%d %d %d\n", len, CMD.header_sz, IMAGE_HEADER_OFFSET);
```
Comment (bigbrett) [inline tools/keytools/sign.c:324]: dead code

File: tools/keytools/sign.c (hunk 3 above)
Comment (philljj) [inline tools/keytools/sign.c:48]: Line should be deleted?

**Review by bigbrett — APPROVED** [review]
Nice, a welcome addition. A few minor nits that can be disregarded

**Review by philljj — DISMISSED** [review]

File: .github/workflows/test-renode-nrf52.yml [hunk 5]
```diff
@@ -23,12 +23,12 @@ jobs:
 # SIGN=NONE TEST
 
       - name: Renode Tests SIGN=NONE
-        run: ./tools/renode/docker-test.sh "SIGN=NONE"
+        run: ./tools/renode/docker-test.sh "SIGN=NONE V=1"
```
[the remark this answers is not in the document]
Comment (dgarske) [reply .github/workflows/test-renode-nrf52.yml:26]: Better logging. Leaving it. There are other cases we use V=1

File: tools/keytools/keygen.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgarske) [reply tools/keytools/keygen.c:460]: Changed to default o level 2. Matches older behavior.

**Review by bigbrett — DISMISSED** [review]

**Review by dgarske — APPROVED** [review]

**Review by bigbrett — APPROVED** [review]

**Pull request merged**